### DISCUSSION TITLE
C++ interface - Any ideas why freerdp_new returns null?

### DISCUSSION THREAD

**Discussion by lupuleasa-core** [FreeRDP/FreeRDP#8104] [Q&A] [answered]
Hi,

We have upgraded freerdp in a legacy product, and have loaded feerdp2.dll instead of the old wfreerdp.dll in our c++ code, but we noticed that calling the freerdp_new() function seems to return null no matter what we do.

Any suggestions on what might be the issue?

We have spent the last month on this, and were wondering if we are missing anything.

**Comment by hardening** [comment]
Are you running last stable-2.0 version or master ?

**Comment by lupuleasa-core** [comment]
@hardening 

Release 2.7.0

Built it with this command, with an extra modification to produce wfreerdp.dll instead of .exe
```
cmake -GNinja -B./build -H./ -DCMAKE_INSTALL_PREFIX=C:\dev\freerdp -DCMAKE_BUILD_TYPE=Release -DCHANNEL_URBDRC=OFF
cmake --build build --target install
```
deployed the .dll's and the headers in our codebase, and only loaded freerdp2.dll (which I saw contains the functions we use like freerdp_new using dumpbin tool)

**Reply by hardening** [reply]
And which toolchain are you using, msvc or mingw ?

**Reply by lupuleasa-core** [reply]
visual c++
I have used the 2019 cmdline version, but I have 2022 installed as well
Our legacy product is built with 2019 version, and it's windows 32 bit only

**Reply by hardening** [reply]
That's really strange, the only explanation would be that `calloc()` returns NULL in `freerdp_new()`.

**Reply by lupuleasa-core** [reply]
As an update: 2.8.0 version latest test we see freerdp_new return non-null, even though we still see the crash we were having
we will investigate

**Reply by hardening** [reply]
Last time I've seen that it was when compiling with mingw and the root cause was a mix of C runtimes (anyway the bug was happening on free, not calloc).

**Reply by lupuleasa-core** [reply]
we found the issue eventually
we were building our product in debug mode and testing it on a release deployment
building our module in release mode fixed the issue

**Comment by lupuleasa-core** [comment, marked as answer]
we found the issue eventually
we were building our product in debug mode and testing it on a release deployment
building our module in release mode fixed the issue
